app/controllers/devise/checkga_controller.rb

```diff
@@ -33,18 +33,22 @@ def update
         end
       else
         set_flash_message(:error, :error)
-        redirect_to :root
+        redirect_to redirect_on_error_url
       end
 
     else
       set_flash_message(:error, :error)
-      redirect_to :root
+      redirect_to redirect_on_error_url
     end
   end
 
+  def redirect_on_error_url
+    @redirect_on_error_url ||= :root
```

Comment:
No need to cache this is there? Configuration is done by overriding anyway, so they'll just return a different string.